ibc-core/ics24-host/types/src/error.rs

```diff
@@ -58,10 +59,12 @@ pub enum IdentifierError {
     InvalidLength { actual: String, min: u64, max: u64 },
     /// id `{0}` can only contain alphanumeric characters or `.`, `_`, `+`, `-`, `#`, - `[`, `]`, `<`, `>`
     InvalidCharacter(String),
-    /// invalid prefix: `{0}`
+    /// invalid prefix `{0}`
     InvalidPrefix(String),
-    /// failed to parse `{value}`: `{description}`
-    FailedToParse { value: String, description: String },
+    /// failed to parse: `{description}`
+    FailedToParse { description: String },
+    /// mismatched event kind: expected {expected}, actual {actual}
+    MismatchedEventKind { expected: String, actual: String },
```

Comment:
I would categorize this under `DecodingError` if that makes sense to you.

Reply:
From what I see, `FailedToParse` is only used when parsing an identifier fails. So, I think it's fine as is.